quick-protobuf/examples/pb_rs_v3/data_types.rs

```diff
@@ -168,75 +210,78 @@ impl<'a> MessageRead<'a> for FooMessage<'a> {
+        + if self.f_sfixed32 != 0i32 { 1 + 4 } else { 0 }
+        + if self.f_double != 0f64 { 1 + 8 } else { 0 }
+        + if self.f_float != 0f32 { 1 + 4 } else { 0 }
+        + if self.f_bytes != Cow::Borrowed(b"") { 1 + sizeof_len((self.f_bytes).len()) } else { 0 }
+        + if self.f_string != Cow::Borrowed("") { 2 + sizeof_len((self.f_string).len()) } else { 0 }
+        + if self.f_self_message.is_some() { 2 + sizeof_len((*(self.f_self_message.as_ref().unwrap())).get_size()) } else { 0 }
+        + 2 + sizeof_len((self.f_bar_message).get_size())
+        + if self.f_repeated_int32.is_empty() { 0 } else { 2 + sizeof_len(self.f_repeated_int32.iter().map(|&s| sizeof_varint((s) as u64)).sum::<usize>()) }
+        + if self.f_repeated_packed_int32.is_empty() { 0 } else { 2 + sizeof_len(self.f_repeated_packed_int32.iter().map(|&s| sizeof_varint((s) as u64)).sum::<usize>()) }
         + if self.f_repeated_packed_float.is_empty() { 0 } else { 2 + sizeof_len(self.f_repeated_packed_float.len() * 4) }
-        + self.f_imported.as_ref().map_or(0, |m| 2 + sizeof_len((m).get_size()))
-        + self.f_baz.as_ref().map_or(0, |m| 2 + sizeof_len((m).get_size()))
-        + self.f_nested.as_ref().map_or(0, |m| 2 + sizeof_len((m).get_size()))
-        + if self.f_nested_enum == data_types::mod_BazMessage::mod_Nested::NestedEnum::Foo { 0 } else { 2 + sizeof_varint(*(&self.f_nested_enum) as u64) }
-        + self.f_map.iter().map(|(k, v)| 2 + sizeof_len(2 + sizeof_len((k).len()) + sizeof_varint(*(v) as u64))).sum::<usize>()
+        + 2 + sizeof_len((self.f_imported).get_size())
+        + 2 + sizeof_len((self.f_baz).get_size())
+        + 2 + sizeof_len((self.f_nested).get_size())
+        + if self.f_nested_enum != data_types::mod_BazMessage::mod_Nested::NestedEnum::Foo { 2 + sizeof_varint((self.f_nested_enum) as u64) } else { 0 }
+        + self.f_map.iter().map(|(k, &v)| 2 + sizeof_len(2 + sizeof_len((k).len()) + sizeof_varint((v) as u64))).sum::<usize>()
         + self.f_repeated_string.iter().map(|s| 2 + sizeof_len((s).len())).sum::<usize>()
         + self.f_repeated_baz_message.iter().map(|s| 2 + sizeof_len((s).get_size())).sum::<usize>()
-        + match self.test_oneof {
-            mod_FooMessage::OneOftest_oneof::f1(ref m) => 2 + sizeof_varint(*(m) as u64),
-            mod_FooMessage::OneOftest_oneof::f2(ref m) => 2 + sizeof_varint(*(m) as u64),
+        + match &self.test_oneof {
+            mod_FooMessage::OneOftest_oneof::f1(m) => 2 + sizeof_varint((*m) as u64),
+            mod_FooMessage::OneOftest_oneof::f2(m) => 2 + sizeof_varint((*m) as u64),
             mod_FooMessage::OneOftest_oneof::f3(ref m) => 2 + sizeof_len((m).len()),
             mod_FooMessage::OneOftest_oneof::None => 0,
-    }    }
+        }
+    }
 
     fn write_message<W: WriterBackend>(&self, w: &mut Writer<W>) -> Result<()> {
-        if self.f_int32 != 0i32 { w.write_with_tag(8, |w| w.write_int32(*&self.f_int32))?; }
```

Comment:
The diff in the generated code would be easier to review if this wouldn't be part of this PR :)

Reply:
Yeah I think my rewriting some of the codegen led to me changing some details without realizing; let me check that the original version still plays nicely with the rest of the generated code. If so this will probably come under a tidy-up PR.